aws/resource_aws_guardduty_filter.go

```diff
@@ -0,0 +1,493 @@
+)
+
+func resourceAwsGuardDutyFilter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGuardDutyFilterCreate,
+		Read:   resourceAwsGuardDutyFilterRead,
+		Update: resourceAwsGuardDutyFilterUpdate,
+		Delete: resourceAwsGuardDutyFilterDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
+			"finding_criteria": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"criterion": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"field": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(criteriaFields(), false),
+									},
+									"condition": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"equals",
+											"not_equals",
+											"greater_than",
+											"greater_than_or_equal",
+											"less_than",
+											"less_than_or_equal",
+										}, false),
+									},
+									"values": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"NOOP",
+					"ARCHIVE",
+				}, false),
+			},
+			"rank": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Second),
+			Update: schema.DefaultTimeout(60 * time.Second),
+		},
+	}
+}
+
+func resourceAwsGuardDutyFilterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	input := guardduty.CreateFilterInput{
+		Action:      aws.String(d.Get("action").(string)),
+		Description: aws.String(d.Get("description").(string)),
+		DetectorId:  aws.String(d.Get("detector_id").(string)),
+		Name:        aws.String(d.Get("name").(string)),
+		Rank:        aws.Int64(int64(d.Get("rank").(int))),
+	}
+
+	// building `FindingCriteria`
+	findingCriteria := d.Get("finding_criteria").([]interface{})[0].(map[string]interface{})
+
+	var err error
+	input.FindingCriteria, err = serializeFindingCriteria(findingCriteria)
+	if err != nil {
+		return err
+	}
+
+	tagsInterface := d.Get("tags").(map[string]interface{})
```

Comment:
Please update this to use the new `keyvaluetags` package at https://github.com/terraform-providers/terraform-provider-aws/tree/master/aws/internal/keyvaluetags